tools/tpm2_load.c

```diff
@@ -54,6 +89,184 @@ static tool_rc load(ESYS_CONTEXT *ectx) {
+    return false;
+}
+
+char getChar(int value) {
+    if (value >=0 && value <= 9) {
+        return ((char) value) + '0';
+    } else if (value >= 10 && value <= 15) {
+        return ((char) value) + 'a';
+    } else {
+        LOG_ERR("Unable to create parent using create primary");
+        return 'Z';
+    }
+}
+
+static const TPM2B_SENSITIVE_CREATE primarySensitive = {
+    .sensitive = {
+        .userAuth = {
+             .size = 0,
+         },
+        .data = {
+             .size = 0,
+         }
+    }
+};
+
+static const TPM2B_DATA allOutsideInfo = {
+    .size = 0,
+};
+
+static const TPML_PCR_SELECTION allCreationPCR = {
+    .count = 0,
+};
+
+/*static const TPM2B_PUBLIC primaryRsaTemplate = {
+    .publicArea = {
+        .type = TPM2_ALG_RSA,
+        .nameAlg = TPM2_ALG_SHA256,
+        .objectAttributes = (TPMA_OBJECT_USERWITHAUTH |
+                             TPMA_OBJECT_RESTRICTED |
+                             TPMA_OBJECT_DECRYPT |
+                             TPMA_OBJECT_NODA |
+                             TPMA_OBJECT_FIXEDTPM |
+                             TPMA_OBJECT_FIXEDPARENT |
+                             TPMA_OBJECT_SENSITIVEDATAORIGIN),
+        .authPolicy = {
+             .size = 0,
+         },
+        .parameters.rsaDetail = {
+             .symmetric = {
+                 .algorithm = TPM2_ALG_AES,
+                 .keyBits.aes = 128,
+                 .mode.aes = TPM2_ALG_CFB,
+              },
+             .scheme = {
+                .scheme = TPM2_ALG_NULL,
+                .details = {}
+             },
+             .keyBits = 2048,
+             .exponent = 0,
+         },
+        .unique.rsa = {
+             .size = 0,
+         }
+     }
+};*/
+
+static const TPM2B_PUBLIC primaryEccTemplate = {
+    .publicArea = {
+        .type = TPM2_ALG_ECC,
+        .nameAlg = TPM2_ALG_SHA256,
+        .objectAttributes = (TPMA_OBJECT_USERWITHAUTH |
+                             TPMA_OBJECT_RESTRICTED |
+                             TPMA_OBJECT_DECRYPT |
+                             TPMA_OBJECT_NODA |
+                             TPMA_OBJECT_FIXEDTPM |
+                             TPMA_OBJECT_FIXEDPARENT |
+                             TPMA_OBJECT_SENSITIVEDATAORIGIN),
+        .authPolicy = {
+             .size = 0,
+         },
+        .parameters.eccDetail = {
+             .symmetric = {
+                 .algorithm = TPM2_ALG_AES,
+                 .keyBits.aes = 128,
+                 .mode.aes = TPM2_ALG_CFB,
+              },
+             .scheme = {
+                .scheme = TPM2_ALG_NULL,
+                .details = {}
+             },
+             .curveID = TPM2_ECC_NIST_P256,
+             .kdf = {
+                .scheme = TPM2_ALG_NULL,
+                .details = {}
+             },
+         },
+        .unique.ecc = {
+             .x.size = 0,
+             .y.size = 0
+         }
+     }
+};
+
```

Comment:
use getcap to determine available algorithms on the TPM. If ECC is supported then always use ECC as parent object, else use RSA as parent object. @williamcroberts do you agree?